### PR TITLE
Basic implementation of `bands.x`

### PR DESCRIPTION
**Pull request by bastonero**
An implementation that wraps the binary `bands.x` was missing. This program is used to find the symmetries of wavefunctions, to re-order the bands, and to perform basic post-processings, such as the calculation of the momentum operator.

The current parser only performs basic parsing, with no specialized outputs for k-point wavefunction symmetries, nor momentum operator, nor bands. This can be, for instance, implemented over time depending on user request.

### CODE REVIEW

File: src/aiida_quantumespresso/calculations/bands.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+"""`CalcJob` implementation for the q2r.x code of Quantum ESPRESSO."""
```
Comment (sphuber) [inline src/aiida_quantumespresso/calculations/bands.py:2]:
```suggestion
"""`CalcJob` implementation for the bands.x code of Quantum ESPRESSO."""
```

File: src/aiida_quantumespresso/calculations/bands.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    """
+
+    _MOMENTUM_OPERATOR_NAME = 'momentum_operator.dat'
+    _BANDS_NAME = 'bands.dat'
+
+    _default_namelists = ['BANDS']
+    _blocked_keywords = [
+        ('BANDS', 'outdir', NamelistsCalculation._OUTPUT_SUBFOLDER),  # pylint: disable=protected-access
+        ('BANDS', 'prefix', NamelistsCalculation._PREFIX),  # pylint: disable=protected-access
+        ('BANDS', 'filband', _BANDS_NAME),  # pylint: disable=protected-access
```
Comment (sphuber) [inline src/aiida_quantumespresso/calculations/bands.py:25]: I think this rule is not used, right?
```suggestion
        ('BANDS', 'filband', _BANDS_NAME),
```

File: tests/parsers/fixtures/bands/default/aiida.out
Comment (sphuber) [inline tests/parsers/fixtures/bands/default/aiida.out:1]: Can you please cut this output to the bare minimum? Just whatever is needed for the parsing to work

**Review by sphuber — CHANGES_REQUESTED** [review]
Thanks @bastonero all looks good. Just some minor requests.

As a side note, not sure if you are already aware of `aiida-shell` but for it makes it really easy to run these small kind of utilities without having to write dedicated plugins. You can see [an example of QE the with use of `bands.x`](https://aiida-shell.readthedocs.io/en/latest/examples/qe.html#id6) in the online docs.

File: src/aiida_quantumespresso/calculations/bands.py (hunk 2 above)
Comment (bastonero) [reply src/aiida_quantumespresso/calculations/bands.py:25]: Right, thanks!

File: tests/parsers/fixtures/bands/default/aiida.out
Comment (bastonero) [reply tests/parsers/fixtures/bands/default/aiida.out:1]: Sure

**Comment by bastonero** [comment]
Thanks a lot @sphuber for the review.

I was aware of the plugin, but thanks for the reminder, I wasn't aware of the very nice docs. Here I was interested to use the CalcJob/WorkChain for future composite WorkChains. But good to keep in mind about aiida-shell.

**Comment by bastonero** [comment]
PS: any idea why the docs is always failing lately?

**Comment by mbercx** [comment]
> PS: any idea why the docs is always failing lately?


Hmm, let me have a look 👀

**Review by sphuber — APPROVED** [review]

**Comment by sphuber** [comment]
Fixed the RTD build so merging this. Thanks @bastonero

**Pull request merged**